experimental/GroebnerWalk/src/generic_walk.jl

```diff
@@ -0,0 +1,245 @@
+@doc raw"""
+    generic_walk(G::Oscar.IdealGens, start::MonomialOrdering, target::MonomialOrdering)
+
+Compute a reduced Groebner basis w.r.t. to a monomial order by converting it using the Groebner Walk
+using the algorithm proposed by Collart, Kalkbrener & Mall (1997).
+
+# Arguments
+- `G::Oscar.IdealGens`: Groebner basis of an ideal with respect to a starting monomial order.
+- `target::MonomialOrdering`: monomial order one wants to compute a Groebner basis for.
+- `start::MonomialOrdering`: monomial order to begin the conversion.
+"""
+function generic_walk(G::Oscar.IdealGens, start::MonomialOrdering, target::MonomialOrdering)
+  @vprintln :groebner_walk "Results for generic_walk"
+  @vprintln :groebner_walk "Facets crossed for: "
+
+  Lm = leading_term.(G, ordering = start)
+  MG = MarkedGroebnerBasis(gens(G), Lm)
+  v = next_gamma(MG, ZZ.([0]), start, target)
+
+  @v_do :groebner_walk steps = 0
+  while !isempty(v)
+    @vprintln :groebner_walk v
+    MG = generic_step(MG, v, target)
+    v = next_gamma(MG, v, start, target)
+
+    @v_do :groebner_walk steps += 1
+    @vprintln :groebner_walk 2 G
+    @vprintln :groebner_walk 2 "======="
+  end
+
+  @vprint :groebner_walk "Cones crossed: "
+  @vprintln :groebner_walk steps
+  return gens(MG)
+end
+
+@doc raw"""
+    generic_step(MG::MarkedGroebnerBasis, v::Vector{ZZRingElem}, ord::MonomialOrdering)
+
+Given the marked Gröbner basis `MG` and a facet normal vector `v`, compute the next marked Gröbner basis.
+"""
+function generic_step(MG::MarkedGroebnerBasis, v::Vector{ZZRingElem}, ord::MonomialOrdering)
+  facet_generators = facet_initials(MG, v)
+  H = groebner_basis(
+    ideal(facet_generators); ordering=ord, complete_reduction=true, algorithm=:buchberger
+  )
+
+  @vprint :groebner_walk 5 "Initials forms of facet: "
+  @vprintln :groebner_walk 5 facet_generators
+
+  @vprint :groebner_walk 3 "GB of initial forms: "
+  @vprintln :groebner_walk 3 H
+
+  H = MarkedGroebnerBasis(gens(H), leading_term.(H; ordering = ord))
+
+  lift_generic!(H, MG)
+  autoreduce!(H)
+
+  return H
+end
+
+@doc raw"""
+    difference_lead_tail(MG::MarkedGroebnerBasis)
+
+Computes $a - b$ for $a$ a leading exponent and $b$ in the tail of some $g\in MG$. 
+"""
+function difference_lead_tail(MG::MarkedGroebnerBasis)
+  (G,Lm) = gens(MG), markings(MG)
+  lead_exp = leading_exponent_vector.(Lm)
+  
+  v = map(zip(lead_exp, exponents.(G))) do (l,T)
+    Ref(l) .- T
```

Comment:
what's the purpose of using `Ref` here? I unfortunately don't see the issue you are trying to work around to think about more julian or more efficient ways to do it.

Reply:
`l` is a Vector and `T` is a Vector of Vector. Since I want to subtract every element from `T` from `l`, I need to fix `l` for broadcasting with `Ref`.